quconot/implementations/mct_barenco_74_dirty.py

```diff
@@ -23,13 +23,7 @@ def __init__(self, controls_no: int, **kwargs) -> None:
 
     def TofDecomp(self):
         qc = QuantumCircuit(3)
-        qc.ry(np.pi / 4, 2)
-        qc.cx(1, 2)
-        qc.ry(np.pi / 4, 2)
-        qc.cx(0, 2)
-        qc.ry(-np.pi / 4, 2)
-        qc.cx(1, 2)
-        qc.ry(-np.pi / 4, 2)
+        qc.ccx(0, 1, 2)

```

Comment:
this is a fundamental part of this implementation. that it uses relative phase toffoli to reduce cost.
i will have to see what the problem really is here